src/system/NewDialog/DialogClose.tsx

```diff
@@ -0,0 +1,58 @@
+/** @jsxImportSource theme-ui */
+
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import React, { forwardRef } from 'react';
+import { IoClose } from 'react-icons/io5';
+import { Theme } from 'theme-ui';
+
+interface DialogCloseProps {
+	children: React.ReactNode;
+}
+
+const DialogClose = forwardRef<
+	React.ElementRef< typeof DialogPrimitive.Close >,
+	DialogCloseProps
+>( ( props, forwardedRef ) => (
+	<DialogPrimitive.Close asChild { ...props } ref={ forwardedRef }>
```

Comment:
This doesn't really make sense. If we are forwarding the ref to the child element, can't the consumer just add the ref themselves?